log-group-events-lambda/handler/cw_ops.go

```diff
@@ -0,0 +1,198 @@
+package handler
+
+import (
+	"errors"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/cloudwatchlogs"
+	"github.com/aws/aws-sdk-go/service/cloudwatchlogs/cloudwatchlogsiface"
+	"github.com/hashicorp/go-multierror"
+	"github.com/logzio/firehose-logs/common"
+	"sync"
+	"time"
+)
+
+type CloudWatchLogsClient struct {
+	Client cloudwatchlogsiface.CloudWatchLogsAPI
+	Mutex  sync.Mutex
+}
+
+func getCloudWatchLogsClient() (*CloudWatchLogsClient, error) {
+	sess, err := common.GetSession()
+	if err != nil {
+		return nil, err
+	}
+	return &CloudWatchLogsClient{Client: cloudwatchlogs.New(sess)}, nil
+}
+
+func (cwLogsClient *CloudWatchLogsClient) addSubscriptionFilter(logGroups []string) ([]string, error) {
+	if cwLogsClient == nil {
+		return nil, fmt.Errorf("CloudWatch Logs client is nil")
+	}
+
+	destinationArn := envConfig.destinationArn
+	roleArn := envConfig.roleArn
+	filterPattern := ""
+	filterName := subscriptionFilterName
+	added := make([]string, 0, len(logGroups))
+	var result *multierror.Error
+
+	var wg sync.WaitGroup
```

Comment:
`updateSubscriptionFilters` method performs both add and delete operations, and we run each of them separately to manage the concurrent calls to AWS (because for large amount of log groups it may be high in each operation). So running them independently helps reduce the overall concurrency, which means each needs it's own wait group